### PR TITLE
Accurate `sizeof` for pandas `string[python]` dtype

### PR DESCRIPTION
**Pull request by crusaderky**
Follow-up to #9776

### CODE REVIEW

File: dask/sizeof.py [hunk 1]
```diff
@@ -135,7 +135,12 @@ def register_pandas():
     import numpy as np
     import pandas as pd
 
-    from dask.dataframe._compat import dtype_eq
+    from dask.dataframe._compat import PANDAS_GT_130, dtype_eq
+
+    if PANDAS_GT_130:
+        OBJECT_DTYPES = (object, pd.StringDtype())
```
Comment (mrocklin) [inline dask/sizeof.py:141]: We wouldn't want to use the special code if the dtype is `string[pyarrow]`.  Will that type equate to `pd.StringDtype()`?  If so, we'll want to be more careful here.  If not, then please ignore.

Comment (crusaderky) [reply dask/sizeof.py:141]: `pd.StringDtype()` is the same as `string[python]`.

**Comment by mrocklin** [comment]
This looks good to me.  @jrbourbeau I defer to you though.

File: dask/sizeof.py (hunk 1 above)
Comment (jrbourbeau) [reply dask/sizeof.py:141]: There's a config option in `pandas` which allows `pd.StringDtype()` to default to `pyarrow` strings

```python
In [1]: import pandas as pd

In [2]: pd.set_option("mode.string_storage", "pyarrow")

In [3]: pd.StringDtype()
Out[3]: string[pyarrow]

In [4]: s = pd.Series(["foo", "bar"], dtype=pd.StringDtype("pyarrow"))

In [5]: s.dtype in (object, pd.StringDtype())
Out[5]: True

In [6]: s = pd.Series(["foo", "bar"], dtype=pd.StringDtype("python"))

In [7]: s.dtype in (object, pd.StringDtype())
Out[7]: False
```

Probably best to explicitly specify that we always want `string[python]` here 

```suggestion
        OBJECT_DTYPES = (object, pd.StringDtype("python"))
```

**Review by jrbourbeau — COMMENTED** [review]
One small suggestion, otherwise this LGTM

**Pull request merged**